### ISSUE TITLE
IMCs don't work

### ISSUE THREAD

**Issue by TheLMiffy1111** [Azanor/thaumcraft-beta#117]
Hello,
I am making an addon that adds native clusters for metallic ores from other mods, and I would like to implement the feature when a ore block is broken with a tool with the Refining infusion enchantment on it, it would drop the corresponding native ore cluster.
I got the dev version and found that there were IMC handlers for this, but they weren't working properly.

**Comment by Azanor** [comment]
Could you expand a bit more on what is not working?

**Comment by TheLMiffy1111** [comment]
While making the addon, I looked into Thaumcraft with a decompiling addon to see if there was any way to implement this feature without referencing Thaumcraft's code directly, and I found IMC handling in Thaumcraft's CommonProxy, which uses "nativeCluster" as the key. I therefore tried adding the feature via IMC, but it did not work. I put a breakpoint over that part of the code, and found that no IMC messages were fetched, despite my mod already sent the IMC messages. I think it was probably because the Object that was used to fetch the messages was the proxy itself, whereas Forge says it has to be an instance of the mod or the mod id.
